airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftCopyDestination.java

```diff
@@ -165,13 +178,43 @@ protected void acceptTracked(AirbyteMessage message) throws Exception {
      */
     @Override
     protected void close(boolean hasFailed) throws Exception {
-      RedshiftCopier.closeAsOneTransaction(new ArrayList<>(tableNameToCopier.values()), hasFailed, null);
+      RedshiftCopier.closeAsOneTransaction(new ArrayList<>(streamNameToCopier.values()), hasFailed, redshiftDb);
     }
 
   }
 
-  public static void main(String[] args) throws Exception {
-    new IntegrationRunner(new RedshiftCopyDestination()).run(args);
+  public static class S3Config {
+
+    public String bucketName;
+    public String region;
+    public String accessKeyId;
+    public String secretAccessKey;
+
+    public S3Config(JsonNode config) {
+      this.bucketName = config.get("s3_bucket_name").asText();
+      this.region = config.get("s3_bucket_region").asText();
+      this.accessKeyId = config.get("access_key_id").asText();
+      this.secretAccessKey = config.get("secret_access_key").asText();
+    }
+
+    public static boolean isPresent(JsonNode config) {
+      var bucketNode = config.get("s3_bucket_name");
+      var regionNode = config.get("s3_bucket_region");
+      var accessKeyIdNode = config.get("access_key_id");
+      var secretAccessKeyNode = config.get("secret_access_key");
+
+      var emptyRegion = regionNode == null || regionNode.asText().equals("");
+
+      if (bucketNode == null && emptyRegion && accessKeyIdNode == null && secretAccessKeyNode == null) {
```

Comment:
should this check be the same for all different params? either they all exist or not? Just asking because in some cases we seem to be doing a null check but for region we seem to be doing a empty string check too.

Reply:
good question. the empty string check is because of how the front end is dealing with the enum json schema type. in order to have a drop down, I had to specify the blank string. I'll drop a comment to make this clear.